lib/cc/yaml/parser/psych.rb

```diff
@@ -80,10 +82,17 @@ def parse(root = nil)
         parsed ||= ::Psych.parse(@value)
         accept(root, parsed)
         root
+        check_for_analysis_key(root)
```

Comment:
I think you've accidentally put this after the method's intended return value (`root`, one line up). This forced you to return `root` from the check method and it makes the statement above a no-op.